pkg/ratelimit/runner.go

```diff
@@ -48,9 +51,9 @@ type Runner interface {
 type Task struct {
 	ctx         context.Context
 	submittedAt time.Time
-	opts        *TaskOpts
 	f           func(context.Context)
 	name        string
+	retained    bool
```

Comment:
This naming is vague without looking at the actual logic, maybe a line of comment would be helpful.